_posts/2024-01-16-fido-passkey.md

```diff
@@ -0,0 +1,1127 @@
+            startRegistration(result).then(result => {
+                this.registerService.verify(this.username, result).subscribe(result => {
+                    type ObjectKey = keyof typeof result;
+
+                    const verifiedVar = 'verified' as ObjectKey;
+                    const verified = result[verifiedVar].toString();
+
+                    if (verified === 'true') {
+                        alert('Registration success.');
+                    } else {
+                        alert('Registration failed.');
+                    }
+                });
+            }, error => {
+                console.error(error);
+                alert('Registration failed.');
+            });
+        });
+    }
+
+    login() {
+        this.loginService.login(this.username).subscribe(result => {
+            startAuthentication(result).then(result => {
+                this.loginService.verify(this.username, result).subscribe(result => {
+                    type ObjectKey = keyof typeof result;
+
+                    const verifiedVar = 'verified' as ObjectKey;
+                    const verified = result[verifiedVar].toString();
+
+                    if (verified === 'false') {
+                        alert('Login failed.');
+                    } else {
+                        this.router.navigateByUrl('/user');
+                    }
+                });
+            }, error => {
+                console.error(error);
+                alert('Login failed.');
+            });
+        });
+    }
+}
+```
+
+Paste the following code in the `home.component.html` file.
+```html
+<div class="container mt-5">
+    <div class="row">
+        <div class="col-sm-12 col-md-8 info-text mt-5">
+            <p>Experience a new era of online security with FIDO Passwordless Authentication. Say goodbye to the hassle of remembering and managing passwords. 
+            Our cutting-edge technology ensures a seamless and secure login process for users, making online interactions effortless and worry-free.</p>
+        </div>
+
+        <div class="col-sm-12 col-md-4">
+            <form>
+                <div class="mb-3">
+                    <label for="username" class="form-label">Username</label>
+                    <input type="text" class="form-control" id="username" aria-describedby="usernameHelp"
+                           [(ngModel)]="username" [ngModelOptions]="{standalone: true}">
+                    <div id="usernameHelp" class="form-text">We'll never share your info with anyone else.</div>
+                </div>
+                <button type="submit" class="btn btn-primary" (click)="register()">Register</button>
+                <button type="submit" class="btn btn-primary ms-2" (click)="login()">Login</button>
+            </form>
+        </div>
+    </div>
+</div>
+```
+
+Paste the following code in the `user.component.html` file.
+```html
+<p>Welcome</p>
+```
```

Comment:
Same here, move the angular code to a github repo and reference it here. Explain what the different components do and why they are needed.